pkg/controller/servicediscovery/servicediscovery_controller.go

```diff
@@ -369,36 +371,52 @@ func updateDNSConfigMap(client controllerClient.Client, k8sclientSet *clientset.
 		    forward . 2.2.2.2:5353
 		}
 		*/
-		corefile := configMap.Data["Corefile"]
-		if strings.Contains(corefile, "lighthouse") {
-			// Assume this means we've already set the ConfigMap up
-			return nil
-		}
 		lighthouseDnsService := &corev1.Service{}
 		err = client.Get(context.TODO(), types.NamespacedName{Name: lighthouseCoreDNSName, Namespace: cr.Namespace}, lighthouseDnsService)
-		if err != nil || lighthouseDnsService.Spec.ClusterIP == "" {
+		lighthouseClusterIp := lighthouseDnsService.Spec.ClusterIP
+		if err != nil || lighthouseClusterIp == "" {
 			return goerrors.New("lighthouseDnsService ClusterIp should be available")
 		}
-		expectedCorefile := `#lighthouse
-clusterset.local:53 {
-forward . `
-		expectedCorefile = expectedCorefile + lighthouseDnsService.Spec.ClusterIP + "\n" + "}\n"
-		superclusterCorefile := `supercluster.local:53 {
-forward . `
-		expectedCorefile = expectedCorefile + superclusterCorefile + lighthouseDnsService.Spec.ClusterIP + "\n" + "}\n"
+
 		coreFile := configMap.Data["Corefile"]
 		if strings.Contains(coreFile, "clusterset") {
 			// Assume this means we've already set the ConfigMap up
-			return nil
+			coreFile := configMap.Data["Corefile"]
+			if strings.Contains(coreFile, "clusterset.local") {
```

Comment:
We are not doing this for supercluster, but it's fine I guess (for KIND deployments)